src/pod-security-policy/seccompv2/constraint.tmpl

```diff
@@ -0,0 +1,85 @@
+apiVersion: templates.gatekeeper.sh/v1
+kind: ConstraintTemplate
+metadata:
+  name: k8spspseccompv2
+  annotations:
+    metadata.gatekeeper.sh/title: "Seccomp V2"
+    metadata.gatekeeper.sh/version: 1.0.0
+    description: >-
+      Controls the seccomp profile used by containers. Corresponds to the
+      `seccomp.security.alpha.kubernetes.io/allowedProfileNames` annotation on
+      a PodSecurityPolicy. For more information, see
+      https://kubernetes.io/docs/concepts/policy/pod-security-policy/#seccompv2
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sPSPSeccompV2
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          type: object
+          description: >-
+            Controls the seccomp profile used by containers. Corresponds to the
+            `seccomp.security.alpha.kubernetes.io/allowedProfileNames` annotation on
+            a PodSecurityPolicy. For more information, see
+            https://kubernetes.io/docs/concepts/policy/pod-security-policy/#seccompv2
+          properties:
+            allowAnnotations:
+              description: >-
+                Allow reading security contexts from annotations. If set to true, the policy will read security contexts from annotations when missing in spec.
+              type: boolean
+              default: true
+            exemptImages:
+              description: >-
+                Any container that uses an image that matches an entry in this list will be excluded
+                from enforcement. Prefix-matching can be signified with `*`. For example: `my-image-*`.
+
+                It is recommended that users use the fully-qualified Docker image name (e.g. start with a domain name)
+                in order to avoid unexpectedly exempting images from an untrusted repository.
+              type: array
+              items:
+                type: string
+            allowedProfiles:
+              type: array
+              description: >-
+                An array of allowed profile values for seccomp on Pods/Containers.
+
```

Comment:
If we're doing a v2, since annotations are removed from K8s, we can make everyone's lives easier by only supporting the new syntax.

This will make the API match K8s's APIs 1:1.

It will also eliminate all need for translation logic, streamlining the code.

If users still need to enforce against annotations, they can keep using v1. Annotation CT will not bitrot, since that API no longer exists.

Reply:
Makes sense, updated v2 to remove reading from annotations altogether.